packages/grid/x-data-grid-pro/src/hooks/features/detailPanel/useGridDetailPanel.ts

```diff
@@ -74,10 +74,19 @@ export const useGridDetailPanel = (
 
   const handleCellKeyDown = React.useCallback<GridEventListener<GridEvents.cellKeyDown>>(
     (params, event) => {
-      if (!event.ctrlKey || event.key !== 'Enter' || props.getDetailPanelContent == null) {
-        return;
+      if (
+        (event.ctrlKey || event.metaKey) &&
+        event.key === 'Enter' &&
+        props.getDetailPanelContent != null
+      ) {
+        apiRef.current.toggleDetailPanel(params.id);
+      }
+
+      if (params.field === GRID_DETAIL_PANEL_TOGGLE_FIELD && event.key === ' ') {
+        apiRef.current.toggleDetailPanel(params.id);
+        event.stopPropagation();
+        event.preventDefault();
```

Comment:
Instead of `event.stopPropagation()`, could you add a check in https://github.com/mui/mui-x/blob/a84a5c52b240422019c49308b25f28b4211e9b0f/packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts#L117-L118?

We will have to duplicate the field identifier but I think it's OK.

Using `event.stopPropagation()` to disable default behaviors should be avoided whenever possible. If a listener is added to a parent element, it won't be called. I think that for #3844 we will have to create a global `keydown` listener, similar to the global `click` listener. https://github.com/mui/mui-x/blob/a84a5c52b240422019c49308b25f28b4211e9b0f/packages/grid/x-data-grid/src/hooks/features/focus/useGridFocus.ts#L266-L267

Any `event.stopPropagation()` will make it to not be called.